### PR TITLE
Ensure bot adds reactions in given order.

### PR DESCRIPTION
**Pull request by CodeMan99**
Listens to the "reaction_added" event as suggested by the [reactions.add docs](https://api.slack.com/methods/reactions.add).

### CODE REVIEW

**Comment by carterbs** [comment]
Can you just use async? It's so much easier for me to think about. and since i have to maintain this guy, it would be nice if it was in async. Also, conflicts.

**Comment by CodeMan99** [comment]
Use async? For events?

Last I checked async is really for node style callbacks. So I'd need to do a lot of extra work to make sure each emit is one callback execution.

**Comment by CodeMan99** [comment]
Also, needs to be tested.

File: classes/LunchBot.js [hunk 1]
```diff
@@ -129,23 +128,42 @@ class LunchBot {
...
+			params.name = position.value;
+			api.callAPI("reactions.add", params, err => {
+				position = iterator.next();
+
+				if (err) {
+					reject(err);
+					position.done = true;
+				}
+			});
+		};
```
Comment (CodeMan99) [inline classes/LunchBot.js]: I can make this function slightly more readable at the cost of needing to create a callback that delegates to resolve/reject in the promise like the following.

```javascript
new Promise((resolve, reject) => {
   const callback = (err, done) => {
      if (err) return reject(err);
      if (done) return resolve();
   };

   // ...
});
```

Do you think that is preferable?

**Review by carterbs — APPROVED** [review]

**Pull request merged**